### PR TITLE
VEN-1117 | Reorganize and refactor customer UI

### PR DESCRIPTION
**Pull request by TatuArvela**
## Description :sparkles:

* Reorganize and refactor customer UI
  * See details in commit messages

### Closes :no_good_woman:

* [VEN-1117](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1117): Reorganize and refactor customer UI

### Automated tests :gear:️

* Updated all relevant tests and browser tests

### Manual testing :construction_worker_man:

* No functional changes

### CODE REVIEW

**Comment by TatuArvela** [comment]
The selected refactoring scope is now done. However, some changes to `master` were merged during the process, so I still need to manually rebase this.

File: src/features/berth/berthPage/berthLegend/applicationSelector/ApplicationSelectorContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+export type ApplicationSelectorContainerProps = {
+  berthsApplicationType: string;
+  className?: string;
+  resetBerthLimit: Function;
+  selectedBerthCount: number;
+  setBerthLimit: Function;
+  switchApplication: Function;
+};
+
+export const _ApplicationSelectorContainer = ({
```
Comment (Nizar-Rahme) [inline src/features/berth/berthPage/berthLegend/applicationSelector/ApplicationSelectorContainer.tsx:25]: Why did you prefix it with an underscore?

**Review by Nizar-Rahme — CHANGES_REQUESTED** [review]
Great work! 💪 
Left a few comments

File: src/features/berth/berthPage/BerthPage.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  deselectBerth: Function;
+  deselectService: Function;
+  filtered: BerthsType;
+  filteredNot: BerthsType;
+  initialValues: BerthFormValues;
+  loading: boolean;
+  localePush: LocalePush;
+  onSubmit: Function;
+  selectBerth: Function;
+  selectService: Function;
```
[the remark this answers is not in the document]
Comment (TatuArvela) [reply src/features/berth/berthPage/BerthPage.tsx:36]: I'll take a look, some functions may be a bit difficult to type

File: src/features/winterStorage/winterStoragePage/WinterStoragePage.tsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
+import React, { useLayoutEffect } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+
+import { getWinterStorageFilterByValues, isResourceSelected } from '../../../common/utils/applicationUtils';
```
[the remark this answers is not in the document]
Comment (TatuArvela) [reply src/features/winterStorage/winterStoragePage/WinterStoragePage.tsx:4]: Good idea 👍  I can move the feature specific ones next to their feature, but some of these are shared

File: src/features/berth/berthPage/berthLegend/applicationSelector/ApplicationSelectorContainer.tsx (hunk 1 above)
Comment (TatuArvela) [reply src/features/berth/berthPage/berthLegend/applicationSelector/ApplicationSelectorContainer.tsx:25]: A named export shouldn't be called the same as the file if it's not the default export, but I'm not sure where I got that from. After removing the underscore ESLint had nothing to say about this

**Review by Nizar-Rahme — APPROVED** [review]
Excellent job! 👍

File: src/common/autoSave/AutoSave.tsx [hunk 4]
```diff
@@ -2,9 +2,9 @@ import React, { Component } from 'react';
 import { FormSpy, FormSpyRenderProps } from 'react-final-form';
 
 interface Props {
-  save: Function;
+  save: (values: any) => void;
   debounce: number;
-  values?: FormData;
+  values?: any;
```
Comment (Nizar-Rahme) [inline src/common/autoSave/AutoSave.tsx:7]: Is it possible to avoid `any` here?

Comment (TatuArvela) [reply src/common/autoSave/AutoSave.tsx]: Not with the current form types, I'm afraid. I did make it generic with `any` and figured out how to make AutoSave work as a function component though. Added new commits for those.

**Review by Nizar-Rahme — APPROVED** [review]

**Pull request merged**